docs/designs/keras_callback.md

```diff
@@ -0,0 +1,208 @@
+# Design to Support Keras Callback
+
+This document describes the design for supporting callback to customize the behavior of model during training, evaluation and inference in ElasticDL.
+
+## Motivation
+
+In deep learning, we generally need to customize the behavior of model during training, evaluation or
+inference, including reading/changing the model. According to the call frequency, we may perform the behavior per batch, per epoch or at the start and end of job. `tf.keras.callbacks.Callback` is an abstract base class and has methods to perform the behavior at different call frequency, such as `on_bath_end`, `on_epoch_end` and so on. So, we adopt the interfaces of `tf.keras.callbacks.Callback` for users to customize the behavior of model in ElasticDL. 
```

Comment:
IMO, we don't use `tf.keras.callbacks.Callback` directly?
Because we add more methods `on_update_batch_end` which are not from `tf.keras.callbacks.Callback`.

Reply:
We can add a method in the subclass of `tf.keras.callbacks.Callback` like:
```python
class TestCallback(tf.keras.callbacks.Callback):
    def on_update_batch_end(self, batch, logs=None):
         ...
```
Although `on_update_batch_end` is not from `tf.keras.callbacks.Callback`, but we can also
call the method independently on the PS not using `CallbacksList`.
However, due to we are developing a high-performance PS using golang, I wonder whether or not we need to use `tf.keras.callbacks.Callback` to define the method for PS. There is a question of how PS using Golang call the method of callbacks defined using python.

Reply:
In the new design, the PS does not call any method of callbacks. So, we will not add `on_update_batch_end`.